GNU utilities are needed to run the tests. Not suitable for upstream.

--- vim-9.0.0823/src/testdir/Makefile
+++ vim-9.0.0823/src/testdir/Makefile
@@ -12,7 +12,7 @@ SCRIPTSOURCE = ../../runtime
 # Comment out this line to see the verbose output of tests.
 #
 # Catches SwapExists to avoid hanging at the ATTENTION prompt.
-REDIR_TEST_TO_NULL = --cmd 'au SwapExists * let v:swapchoice = "e"' | LC_ALL=C LANG=C LANGUAGE=C awk '/Executing Test_/{match($$0, "([0-9][0-9]:[0-9][0-9] *)?Executing Test_[^\\)]*\\)"); print substr($$0, RSTART, RLENGTH) "\r"; fflush()}'
+REDIR_TEST_TO_NULL = --cmd 'au SwapExists * let v:swapchoice = "e"' | LC_ALL=C LANG=C LANGUAGE=C gawk '/Executing Test_/{match($$0, "([0-9][0-9]:[0-9][0-9] *)?Executing Test_[^\\)]*\\)"); print substr($$0, RSTART, RLENGTH) "\r"; fflush()}'
 
 # Uncomment this line to use valgrind for memory leaks and extra warnings.
 #   The output goes into a file "valgrind.testN"
--- vim-9.0.2009/src/testdir/test_terminal.vim
+++ vim-9.0.2009/src/testdir/test_terminal.vim
@@ -515,7 +515,7 @@ call writefile([
 	\ 'set shell=/bin/sh noruler',
 	\ 'terminal',
 	\ 'sleep 200m',
-	\ 'call feedkeys("tail -n 100 -f Xtext\<CR>", "xt")',
+	\ 'call feedkeys("gtail -n 100 -f Xtext\<CR>", "xt")',
 	\ 'sleep 100m',
 	\ 'call feedkeys("\<C-W>N", "xt")',
 	\ ], 'XTest_postponed', 'D')
